src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -388,11 +388,22 @@ boolean refersToImpl(T obj) {
     private native boolean refersTo0(Object o);
 
     /**
-     * Clears this reference object.  Invoking this method will not cause this
-     * object to be enqueued.
+     * Clears this reference object. Invoking this method does not enqueue this
+     * object, and the garbage collector will no longer enqueue this object once
+     * the referent reaches the designated reachability level.
+     * <p>
+     * There is a potential race condition with the garbage collector. When this
+     * method is called, the garbage collector may already be in the process of
+     * (or already completed) clearing and/or enqueueing this reference.
+     *
+     * Avoid this race by ensuring the referent remains strongly-reachable until
+     * after the call to clear(), using {@link #reachabilityFence(Object)} if necessary.
+     *
      *
      * <p> This method is invoked only by Java code; when the garbage collector
-     * clears references it does so directly, without invoking this method.
+     * clears references it does so directly, without invoking this method. The
+     * {@link #enqueue} method also clears references directly, without invoking
```

Comment:
The last sentence might make sense to have as an apiNote? 🤔

Reply:
I will take that sentence out here, and instead update the `enqueue()` doc to say that the GC clears and enqueues references without calling `clear()` or `enqueue()`.